compiler/src/org.graalvm.compiler.hotspot/src/org/graalvm/compiler/hotspot/meta/HotSpotSnippetReflectionProvider.java

```diff
@@ -150,52 +122,23 @@ public Class<?> originalClass(ResolvedJavaType type) {
         return runtime().getMirror(type);
     }
 
-    private static Method lookupMethod(Class<?> declaringClass, String methodName, Class<?>... parameterTypes) throws ReflectiveOperationException {
-        Method result = declaringClass.getDeclaredMethod(methodName, parameterTypes);
-        result.setAccessible(true);
-        return result;
-    }
-
     @Override
     public Executable originalMethod(ResolvedJavaMethod method) {
+        Objects.requireNonNull(method);
+        GraalError.guarantee(method instanceof HotSpotResolvedJavaMethod, "Unexpected implementation class: %s", method.getClass());
+
         if (method.isClassInitializer()) {
             /* <clinit> methods never have a corresponding java.lang.reflect.Method. */
             return null;
         }
-
-        if (hotSpotJDKReflectionGetMethod == null) {
-            return null;
-        }
-        try {
-            return (Executable) hotSpotJDKReflectionGetMethod.invoke(null, method);
-        } catch (ReflectiveOperationException ex) {
-            throw rethrow(ex.getCause());
-        }
+        return runtime().getMirror(method);
```

Comment:
This results in:

```
/home/runner/work/mandrel/mandrel/mandrel/compiler/src/org.graalvm.compiler.hotspot/src/org/graalvm/compiler/hotspot/meta/HotSpotSnippetReflectionProvider.java:134: error: incompatible types: ResolvedJavaMethod cannot be converted to ResolvedJavaType
        return runtime().getMirror(method);
```

When building with JDK 17 (both OpenJDK and LabsJDK).

See https://github.com/graalvm/mandrel/actions/runs/3499526212/jobs/5861191268#step:8:445 and https://github.com/graalvm/mandrel/actions/runs/3499602031/jobs/5861349246#step:7:330, respectively.